model/MessageQueue.go

```diff
@@ -0,0 +1,50 @@
+package model
+
+import (
+	"container/heap"
+	"sync"
+
+	"github.com/puchkovki/distr-model/model"
+	"github.com/puchkovki/distr-model/priorityq"
+)
+
+// A MessageQueue is a PriorityQueue structure with mutex for parallel usage.
+type MessageQueue struct {
+	queue priorityq.PriorityQueue
+	mutex sync.Mutex
+}
+
+// Initialiaze establishes the heap invariants.
+func (mq *MessageQueue) Initialiaze() {
+	heap.Init(&mq.queue)
+}
+
+// Dequeue removes and returns the object of the priority queue with the minimum priority.
+func (mq *MessageQueue) Dequeue() (ret model.Message) {
+	mq.mutex.Lock()
+	defer mq.mutex.Unlock()
+
+	// Следует ли отлавливать ошибку вида: доступ к пустой очереди?
```

Comment:
No Russian.
And let the answer be "No".

Reply:
Fixed